src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -144,34 +190,55 @@ internal Binder TypeSyntaxBinder
                 return new SourceLocalSymbol(containingSymbol, scopeBinder, allowRefKind, typeSyntax, identifierToken, declarationKind);
             }
 
-            return new LocalWithInitializer(containingSymbol, scopeBinder, typeSyntax, identifierToken, initializer, initializerBinderOpt, declarationKind);
+            return new LocalWithInitializer(containingSymbol, scopeBinder, typeSyntax, identifierToken, initializer, initializerBinderOpt ?? scopeBinder, declarationKind);
         }
 
+        /// <summary>
+        /// Make a local variable for an out variable declaration whose type is inferred as a side-effect of binding the enclosing expression.
+        /// </summary>
         /// <param name="containingSymbol"></param>
-        /// <param name="scopeBinder">
-        /// Binder that owns the scope for the local, the one that returns it in its <see cref="Binder.Locals"/> array.
-        /// </param>
-        /// <param name="enclosingBinderOpt">
-        /// Enclosing binder for the location where the local is declared, if different from the <paramref name="scopeBinder"/>.
-        /// It should be used to bind something at that location.
-        /// </param>
-        /// <param name="typeSyntax"></param>
+        /// <param name="scopeBinder">The binder for the scope of the local</param>
+        /// <param name="invocationBinderOpt">The binder for the invocation, if different from scopeBinder</param>
+        /// <param name="typeSyntax">The type syntax</param>
         /// <param name="identifierToken"></param>
-        /// <param name="context"></param>
-        public static SourceLocalSymbol MakeOutVariable(
+        /// <param name="declarationKind"></param>
+        /// <param name="context">The expression to be bound, which will result in the variable receiving a type</param>
+        /// <returns></returns>
+        public static SourceLocalSymbol MakeVariableDeclaredInExpression(
             Symbol containingSymbol,
             Binder scopeBinder,
-            Binder enclosingBinderOpt,
+            Binder invocationBinderOpt,
             TypeSyntax typeSyntax,
             SyntaxToken identifierToken,
-            CSharpSyntaxNode context)
+            LocalDeclarationKind declarationKind,
```

Comment:
It is currently inlined into the code you asked about, above, to avoid the redundant IsVar test (three tests of that condition is a bit much). I expect this to be used more in the future as we add further support for decomposition.

I will fix the doc comment to clarify the purpose of this method.

---
In reply to: [75772660](https://github.com/dotnet/roslyn/pull/13274#discussion_r75772660) [](ancestors = 75772660,75772415,75768657,75768189,75756780)